workflow-core/src/main/java/com/squareup/workflow1/Tracing.kt

```diff
@@ -0,0 +1,85 @@
+package com.squareup.workflow1
+
+/**
+ * Tracing facade to install later.
+ *
+ * This follows the static service locator that Timber demonstrates.
+ * See [Timber][https://github.com/JakeWharton/timber/blob/trunk/timber/src/main/java/timber/log/Timber.kt].
+ */
+public class Tracing private constructor() {
+
+  public interface Tracer {
+    public fun beginIdSection(label: () -> WorkflowIdentifier)
+    public fun endIdSection(label: () -> WorkflowIdentifier)
+    public fun beginSection(label: () -> String)
+    public fun endSection(label: () -> String)
+  }
+
+  public companion object : Tracer {
+
+    /**
+     * Note that by default we have nothing installed so nothing gets run! Only install something
+     * in an app you want to profile.
+     */
+    @JvmStatic public fun install(tracer: Tracer) {
```

Comment:
note: from an API standpoint, I generally avoid having the "install" and "log" public APIs in the same place. Install is called once and you forget about it. trace is called many times from different places, and when you trace you don't want to see noise about install.

Reply:
See `logcat {}` for an example of separating out install.